analytics/uptime_data.go

```diff
@@ -1,20 +1,333 @@
 package analytics
 
-import "time"
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/fatih/structs"
+	"gorm.io/gorm"
+)
 
 type UptimeReportData struct {
-	URL          string
-	RequestTime  int64
-	ResponseCode int
-	TCPError     bool
-	ServerError  bool
-	Day          int
-	Month        time.Month
-	Year         int
-	Hour         int
-	Minute       int
-	TimeStamp    time.Time
-	ExpireAt     time.Time `bson:"expireAt" json:"expireAt"`
-	APIID        string
-	OrgID        string
+	URL          string     `json:"url"`
+	RequestTime  int64      `json:"request_time"`
+	ResponseCode int        `json:"response_code"`
+	TCPError     bool       `json:"tcp_error"`
+	ServerError  bool       `json:"server_error"`
+	Day          int        `json:"day"`
+	Month        time.Month `json:"month"`
+	Year         int        `json:"year"`
+	Hour         int        `json:"hour"`
+	Minute       int        `json:"minute"`
+	TimeStamp    time.Time  `json:"timestamp"`
+	ExpireAt     time.Time  `bson:"expireAt"`
+	APIID        string     `json:"api_id"`
+	OrgID        string     `json:"org_id"`
+}
+
+type UptimeReportAggregateSQL struct {
+	ID string `gorm:"primaryKey"`
+
+	Counter `json:"counter" gorm:"embedded"`
+
+	TimeStamp      int64  `json:"timestamp" gorm:"index:dimension, priority:1"`
+	OrgID          string `json:"org_id" gorm:"index:dimension, priority:2"`
+	Dimension      string `json:"dimension" gorm:"index:dimension, priority:3"`
+	DimensionValue string `json:"dimension_value" gorm:"index:dimension, priority:4"`
+
+	Code1x  int `json:"code_1x"`
```

Comment:
Why do we put them separately ? Instead what if we create a map and get rid of `ProcessStatusCodes` etc.?

Reply:
@furkansenharputlu I don't follow you here :(

Reply:
I think @furkansenharputlu was about to remove all `Codexxx` and make it as one property in struct but as map, this makes sense to me

Reply:
Thanks @vaske! I meant exactly same.

Reply:
@furkansenharputlu @vaske  those `code_xx` are **counters columns** in the SQL database, that's why they are like these. if we go with a map structure, we're going to miss that or Am I missing something?

Reply:
Can't we create a code struct and use that? We are using same structure for `SQLAnalyticsRecordAggregate` and `ProcessStatusCodes` is the same function redundantly. Also, `UptimeReportAggregateSQL` and `SQLAnalyticsRecordAggregate` look like same, why do we create a new struct like this?

Reply:
@furkansenharputlu I tried to embed  `SQLAnalyticsRecordAggregate` since it has similar behavior + fields but for some reason, gorm is not using the `gorm:"embedded"` tag and it always adds the struct name as a prefix. For example, for the Counter struct, it adds `counter_XXXX` in each column.

I think the best is to let it like this at this moment and not change everything (tests + dashboard) for this.